subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformerInvoker.java

```diff
@@ -362,43 +408,31 @@ public void persistResult(ImmutableSortedMap<String, CurrentFileCollectionFinger
                 executionHistoryStore.store(
                     identityString,
                     originMetadata,
-                    implementationSnapshot,
+                    context.getImplementationSnapshot(),
                     ImmutableList.of(),
-                    inputSnapshots,
-                    inputFileFingerprints,
+                    context.getInputSnapshots(),
+                    context.getInputFileFingerprints(),
```

Comment:
I'm not sure why moving these around was necessary. The end game here is that these values are to be passed via an increasingly specialized `Context` object instead. We should move in that direction instead.

Reply:
Yes, so they will need to be extracted from the unit of work anyway. That is what I did here - instead of keeping the calculations scattered over the transformer execution, I extracted the whole code to do the work out of the execution, so it can move to the context when we get there.